core/src/test/java/kafka/server/share/SharePartitionManagerTest.java

```diff
@@ -133,7 +133,7 @@ public void testNewContextReturnsFinalContext() {
         SharePartitionManager sharePartitionManager = SharePartitionManagerBuilder.builder().build();
 
         ShareFetchMetadata newReqMetadata = new ShareFetchMetadata(Uuid.ZERO_UUID, -1);
-        ShareFetchContext shareFetchContext = sharePartitionManager.newContext("grp", Collections.emptyMap(), Collections.emptyList(), newReqMetadata);
+        ShareFetchContext shareFetchContext = sharePartitionManager.newContext("grp", Collections.emptyMap(), Collections.emptyList(), newReqMetadata, true);
```

Comment:
Thanks for the review ! During the invocation of `sharePartitionManager.newContext` we don't pass any acknowledgements, so there's no way to know whether acknowledgements are present or not, only the variable `isAcknowledgeDataPresent` provides that information. My understanding says that in the general case, only the first fetch request (with request epoch 0) will not contain any acknowledgements, but all the subsequent requests would. Going by that logic, I have set this variable to false in case request epoch is 0, and true in all the other cases.